### PR TITLE
fix stream binance bug due to homeCurrencies

### LINKED ISSUE #357: Binance Streaming exchange complains about multiple connections
**thisiscam** on Discord is having trouble with staying connected to the Binance websocket. They upgraded XChange to 5.0.6 but otherwise are on the main branch.

```
2021-03-04 02:01:00.036  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Binance: UnsupportedOperationException Exchange only handles a single connection - disconnect the current connection. x 19
2021-03-04 02:01:00.065  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Binance: UpFrontSubscriptionRequiredException Binance exchange only supports up front subscriptions - subscribe at connect time x 1
2021-03-04 02:02:00.003  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Noncritical error summary: [Exception name]: [Error message] x [Count]
2021-03-04 02:02:00.038  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Binance: UnsupportedOperationException Exchange only handles a single connection - disconnect the current connection. x 20
2021-03-04 02:03:00.005  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Noncritical error summary: [Exception name]: [Error message] x [Count]
2021-03-04 02:03:00.035  INFO 98329 --- [   scheduling-1] c.r.arbitrader.service.TradingScheduler  : Binance: UnsupportedOperationException Exchange only handles a single connection - disconnect the current connection. x 20
```

One possibility is the fact that we don't explicitly disconnect subscriptions. In `StreamingTickerStrategy#getTickers()` we connect and add the subscription(s) to a List, but if we ever got in there a second time we just throw away the old ones and replace them instead of actually calling `dispose()` on them first. I'm not sure how we'd do it twice since we don't ever remove anything from the `tickers` Map.

Another possibility is that we're putting the wrong `CurrencyPair`s in the `ProductSubscription` in `StreamingTickerStrategy#getTickers()` due to the `homeCurrency` setting. The code in the `BinanceStreamingMarketDataService` checks to see if the ticker you're asking for is in the original list created when the connection was first connected. If, for example, we put "BTC/USD" in at that point and "BTC/USDT" in later, or the reverse, that could cause this problem too.

### PR DESCRIPTION
**Pull request by thisiscam**
This fixes #357 .

Quote from @scionaltera, the cause is basically:
> Another possibility is that we're putting the wrong CurrencyPairs in the ProductSubscription in StreamingTickerStrategy#getTickers() due to the homeCurrency setting. The code in the BinanceStreamingMarketDataService checks to see if the ticker you're asking for is in the original list created when the connection was first connected. If, for example, we put "BTC/USD" in at that point and "BTC/USDT" in later, or the reverse, that could cause this problem too.

Also upgraded to 5.0.6 so might be better merged after #354 .

### CODE REVIEW

File: src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void getTickers(Exchange stdExchange, List<CurrencyPair> currencyPairs, T
         if (!tickers.containsKey(exchange)) { // we're not receiving prices so we need to (re)connect
             ProductSubscription.ProductSubscriptionBuilder builder = ProductSubscription.create();
 
-            currencyPairs.forEach(builder::addTicker);
+            currencyPairs.forEach(pair -> { builder.addTicker(exchangeService.convertExchangePair(exchange, pair)); });
```
Comment (scionaltera) [inline src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java:55]: This is a good find. Unfortunately I can't merge this PR until after we're ready to go to 5.0.6 and I won't be able to merge with the extra snapshot repository enabled. If you can reduce the PR to just this one change I could merge it today.

**Review by scionaltera — CHANGES_REQUESTED** [review]

File: src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java (hunk 1 above)
Comment (thisiscam) [reply src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java:55]: Ok thanks. Just let you know that without 5.0.6 at least Binance doesn't work for me.
But I guess this line is separate from that issue.

Comment (thisiscam) [reply src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java:55]: @scionaltera done!

Comment (scionaltera) [reply src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java:55]: The line you fixed will still fix the bug in the code as it is now, and you're still free to use 5.0.6 in your own branch. We just haven't had a chance to test the upgrade to 5.0.6 _by itself_ much yet, and I keep merging things on top of #326 and #359 which is slowing testing on those big PRs. I'd like to get them out the door, bump to our next version and unblock everything else.

**Review by scionaltera — APPROVED** [review]

**Pull request merged**